### PR TITLE
Adde Indices.Segment Endpoint + Spec

### PR DESCRIPTION
**Pull request by philkra**

### CODE REVIEW

**Review by delvedor — APPROVED** [review]
Changes LGTM, but given the tile of the pr I'm not sure you need to do something more.

File: output/typescript/types.ts [hunk 1]
```diff
@@ -3977,7 +3977,7 @@ export interface ShardRecovery {
 }
 
 export interface IndexSegment {
-  shards: Record<string, ShardsSegment>
+  shards: Record<string, ShardsSegment | Array<ShardsSegment>>
```
Comment (Mpdreamz) [inline output/typescript/types.ts:3980]: Can this ever be just `ShardsSegment` and not `Array<ShardsSegment>` ? 

I think we might also want to introduce a `@cleanup("reason")` reason annotation @delvedor @SylvainJuge to start annotating server responses that we think should be more regular?

**Comment by delvedor** [comment]
Can you rebase on master?

File: output/typescript/types.ts (hunk 1 above)
Comment (philkra) [reply output/typescript/types.ts:3980]: I think you pinged the Sylvain on that comment 😄

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**